vpr/src/place/place_constraints.cpp

```diff
@@ -141,3 +141,18 @@ bool cluster_floorplanning_check(ClusterBlockId blk_id, t_pl_loc loc) {
 
     return floorplanning_good;
 }
+
+void check_cluster_constraints_init() {
```

Comment:
Why are we assigning an empty partition (no constraint) to all clustered blocks?  Shouldn't we instead be reading the vpr_constraints file, and applying those constraints (or if there is no constraint, setting the empty constraint)?  We'll want the flow where you have a constraint file and do:

pack in one run
place in another run

to work, but that is only going to work if the second run (which skips packing) reads the constraint file.

Reply:
Ah ok, so load_vpr_constraints should also be called here, and then we call a routine that fills in cluster_constraints accordingly.

Reply:
But what if some atoms are packed together in cluster that have incompatible region constraints according to the constraints XML file?